File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        
        matrix = [[0] * n for _ in range(n)]
        l, r, u, d = 0, n - 1, 0, n - 1
        
        count = 1
        while l <= r and u <= d:
            for i in range(l, r + 1):
                matrix[u][i] = count
                count += 1
            u += 1
            
            for i in range(u, d + 1):
                matrix[i][r] = count
                count += 1
            r -= 1
            
            for i in range(r, l - 1, -1):
                matrix[d][i] = count
                count += 1
            d -= 1
            
            for i in range(d, u - 1, -1):
                matrix[i][l] = count 
                count += 1
            l += 1
            
        return matrix
            
            
            
            

File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.java
class Solution {
    public int[][] generateMatrix(int n) {
        int[][] matrix = new int[n][n];
        int count = 1;
        
        int l = 0;
        int u = 0;
        int d = n - 1;
        int r = n - 1;
        
        while (l <= r && u <= d) {
            for (int i=l; i <= r; i++) {
                matrix[u][i] = count++;
            }
            u += 1;
                
            for (int i=u; i <= d; i++) {
                matrix[i][r] = count++;
            }
            r -= 1;
            
            for (int i=r; i >= l; i--) {
                matrix[d][i] = count++;
            }
            d -= 1;
            
            for (int i=d; i>=u; i--) {
                matrix[i][l] = count++;
            }
            l += 1;
        }
        return matrix;
    }
}